api/proto/config/storage.proto

```diff
@@ -242,3 +242,32 @@ message DisksConfig {
   DisksArrayType array_type = 2;
   repeated DisksConfig children = 3;
 }
+
+
+// SnapshotConfig describes a snapshot for a specific logical
+// volume that must exist on the device. It has a required field
+// volume_uuid for which it was created and UUID for snapshot name in ZFS.
+message SnapshotConfig {
+  // The real name of the snapshot in ZFS. It is the link between
+  // the command and the response to the command. It is assumed
+  // that the field will always be filled with a unique value
+  // that the controller will generate.
+  string uuid = 1;
+  // Display name (User-friendly name). This name does not affect
+  // the snapshot properties in ZFS in any way.
+  // Can be filled on the cloud side.
+  // If a snapshot has already been created and this field has changed,
+  // it can be assumed that the friendly name for this snapshot has
+  // been renamed on the cloud side.
+  string display_name = 2;
+  // Volume ID of the volume this snapshot belongs to. The field is
+  // required for all messages. Must always be filled in on the
+  // cloud side before sending to create snapshot command.
+  string volume_uuid = 3;
+  // The command for the snapshot rollback operation.
+  // Keep in mind that in order to return to any of the
+  // previous snapshots, all intermediate snapshots will be
+  // destroyed in ZFS. These snapshots must be marked as
+  // deleted in the SnapshotInfo message.
+  DeviceOpsCmd rollback = 7;
```

Comment:
Instead of a command with a counter couldn't we have some declarative statement of the form "should be running using snaphot X"? That might imply that the UUID=X for that snapshot will stay around since it is in use. Hmm - I feel I'm missing parts of how ZFS rollback to a snapshot works in general. Is there a good document where I can read about this?

Reply:
> I think that we still need a counter since we can roll back to the same snapshot as many times as we want. We also need to notify about the success of the command execution or errors if any.

Very good point.

> To roll back to a snapshot, we need to know only 2 things, this is the name of the snapshot and the volume to which this snapshot belongs. You can also read about the [rollback command here](https://docs.oracle.com/cd/E19253-01/819-5461/gbcxk/index.html).
>
> Another question is, do we need to use DeviceOpsCmd, or should we replace it with a regular uint32? I thought that specifying the command as DeviceOpsCmd would give a clear understanding that we are expecting a command, and we can predict its behavior.

Using DevOpsCmd makes sense, but it might make sense to wrap it in a new RollbackCmd which will include a DevOpsCmd plus the uuid for the volume and snapshot.

Reply:
@eriknordmark I updated the PR and added a wrapper for the rollback command